components/chainhook-sdk/src/observer/http.rs

```diff
@@ -58,116 +79,76 @@ pub async fn handle_new_bitcoin_block(
-                ctx.try_log(|logger| {
-                    slog::crit!(logger, "Could not shut down event observer: {e}")
-                });
-                std::process::exit(1)
-            }
-        }
-
-        return Json(json!({
-            "status": 500,
-            "result": "unable to retrieve_full_block",
-        }));
-    };
 
     let header = block.get_block_header();
     let block_height = header.block_identifier.index;
     prometheus_monitoring.btc_metrics_block_received(block_height);
     match background_job_tx.lock() {
-        Ok(tx) => {
-            let _ = tx.send(ObserverCommand::ProcessBitcoinBlock(block));
-        }
+        Ok(tx) => match tx.send(ObserverCommand::ProcessBitcoinBlock(block)) {
+            Ok(_) => {}
+            Err(e) => {
+                return error_response(
+                    format!("Unable to send stacks chain event: {}", e.to_string()),
+                    ctx,
+                );
+            }
+        },
         Err(e) => {
-            ctx.try_log(|logger| {
-                slog::warn!(
-                    logger,
-                    "unable to acquire background_job_tx: {}",
-                    e.to_string()
-                )
-            });
-            return Json(json!({
-                "status": 500,
-                "result": "Unable to acquire lock",
-            }));
+            return error_response(
+                format!("unable to acquire background_job_tx: {}", e.to_string()),
+                ctx,
+            );
         }
     };
 
     let chain_update = match indexer_rw_lock.inner().write() {
         Ok(mut indexer) => indexer.handle_bitcoin_header(header, ctx),
         Err(e) => {
-            ctx.try_log(|logger| {
-                slog::warn!(
-                    logger,
-                    "unable to acquire indexer_rw_lock: {}",
-                    e.to_string()
-                )
-            });
-            return Json(json!({
-                "status": 500,
-                "result": "Unable to acquire lock",
-            }));
+            return error_response(
+                format!("Unable to acquire indexer_rw_lock: {}", e.to_string()),
+                ctx,
+            );
         }
     };
 
     match chain_update {
         Ok(Some(chain_event)) => {
             prometheus_monitoring.btc_metrics_block_appended(block_height);
             match background_job_tx.lock() {
-                Ok(tx) => {
-                    let _ = tx.send(ObserverCommand::PropagateBitcoinChainEvent(chain_event));
-                }
+                Ok(tx) => match tx.send(ObserverCommand::PropagateBitcoinChainEvent(chain_event)) {
+                    Ok(_) => {}
+                    Err(e) => {
+                        return error_response(
+                            format!("Unable to send stacks chain event: {}", e.to_string()),
+                            ctx,
+                        );
+                    }
+                },
```

Comment:
Please check the comment above about ignoring success cases and nested statements.